lib/types/btrfs.nix

```diff
@@ -1,4 +1,52 @@
 { config, options, diskoLib, lib, rootMountPoint, parent, device, ... }:
+let
+  swapType = lib.types.submodule ({ config, ... }: {
+    options = {
+      enable = lib.mkEnableOption "swap";
+
+      size = lib.mkOption {
+        type = lib.types.nullOr (lib.types.strMatching "^([0-9]+[KMGTP])?$");
+        default = null;
+        description = "Size of the swap file (e.g. 2G)";
+      };
+
+      path = lib.mkOption {
+        type = lib.types.str;
+        default = "swapfile";
+        description = "Path to the swap file (relative to the mountpoint)";
+      };
+    };
+  });
+
+  swapConfig = { mountpoint, swap }: lib.optional swap.enable {
+    swapDevices = [{
+      device =
+        assert lib.asserts.assertMsg (mountpoint != null) "swap requires mountpoint to be set";
+        "${mountpoint}/${swap.path}";
+    }];
+  };
+
+  subvolSwapCreate = mountpoint: swap:
+    if !swap.enable
+    then ""
+    else
+      assert lib.asserts.assertMsg (swap.size != null) "swap size must be set";
+      ''btrfs filesystem mkswapfile --size ${swap.size} ${mountpoint}/${swap.path}'';
+
+  partitionSwapCreate = device: swap:
+    if !swap.enable
+    then ""
+    else
+      assert lib.asserts.assertMsg (swap.size != null) "swap size must be set";
+      ''
```

Comment:
```suggestion
    lib.optionalString swap.enable ''
```

Reply:
Thanks. Didn't know about `optionalString` :). Will update later today.

Reply:
I have updated to using `optionalString`.